src/shims/tls.rs

```diff
@@ -189,6 +196,29 @@ impl<'tcx> TlsData<'tcx> {
         }
         None
     }
+
+    /// Set that dtors are running for `thread`. It is guaranteed not to change
+    /// the existing values stored in `dtors_running` for this thread. Returns
+    /// `true` if dtors for `thread` are already running.
+    fn set_dtors_running_for_thread(&mut self, thread: ThreadId) -> bool {
+        if self.dtors_running.contains_key(&thread) {
+            true
+        } else {
+            self.dtors_running.insert(
+                thread,
+                RunningDtorsState { last_dtor_key: None }
+            );
```

Comment:
`insert` returns whether the ky is already in the map, does it? Why the extra `contains_key`?

Reply:
To ensure that this is true:
```rust
    /// It is guaranteed not to change
    /// the existing values stored in `dtors_running` for this thread.
```
More specifically, unconditional `insert` would overwrite `last_dtor_key`.

Reply:
Ah I see.
Can you use the entry API then? That's IMO more explicit than doing a double-lookup.